### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by chaosk**

### CODE REVIEW

File: djangocms_pageadmin/admin.py [hunk 1]
```diff
@@ -70,9 +93,14 @@ def state(self, obj):
...
     def url(self, obj):
-        path = obj.page.get_path(obj.language)
-        if path is not None:
-            url = obj.page.get_absolute_url(obj.language)
+        path = obj._path
+        url = None
+        with override(obj.language):
+            if obj.page.is_home:
+                url = reverse("pages-root")
+            if path:
```
Comment (Aiky30) [inline djangocms_pageadmin/admin.py]: Is it intended that even if obj.page.is_home is true that a reverse will be used anyway? OR is path always false if the obj is the homepage?

File: djangocms_pageadmin/admin.py [hunk 2]
```diff
@@ -53,15 +55,36 @@ def get_list_display(self, request):
...
+                )
+                .select_related("created_by", "versionlock")
+                .prefetch_related("content"),
+            )
         )
-        return queryset.prefetch_related("versions")
 
     def get_version(self, obj):
-        return Version.objects.get_for_content(obj)
+        return obj.versions.all()[0]
```
Comment (Aiky30) [inline djangocms_pageadmin/admin.py]: Would it not be better to use .first() here rather than [0]?

File: djangocms_pageadmin/admin.py [hunk 3]
```diff
@@ -53,15 +55,36 @@ def get_list_display(self, request):
     def get_queryset(self, request):
         """Filter PageContent objects by current site of the request.
         """
+        url_subquery = PageUrl.objects.filter(
+            language=OuterRef("language"), page=OuterRef("page")
+        )
+        draft_version_lock_subquery = VersionLock.objects.filter(
```
Comment (Aiky30) [inline djangocms_pageadmin/admin.py]: I think we should describe the reason that this is here to help anyone reading this in the future. "Collect locked status to handle  the requirement that lock on a draft version dictates the unpublish permission on a published version".

File: djangocms_pageadmin/admin.py (hunk 2 above)
Comment (chaosk) [reply djangocms_pageadmin/admin.py]: `.first()` bypasses prefetch_related cache, needs to use `.all()[0]`

**Pull request merged**